### DISCUSSION TITLE
Working with worksheets with large amounts of data is very slow

### DISCUSSION THREAD

**Discussion by kungyuu** [dfinke/ImportExcel#1378] [General]
It is very slow to import large data volume excel using import-excel (test data 1 worksheet with 16 columns and 600,000 rows of data), version 7.4.1 takes about 3 minutes, the same logic, using version 7.8.4 does not work properly. The script is as follows:[System.IO.File]::WriteAllText($out_path, -join($(foreach ($r in $(Import-Excel -Path $in_path -HeaderName ('a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', ' p') -StartRow 2)) {($r.psobject.properties.value -join "|") + "`n"})))。 However, with Python's Pandas module, it can be done in less than 2 minutes. Python script: import pandas as pd
pd.read_excel(in_path, sheet_name=0).to_csv(out_path, sep=sep, header=0, index=0)

**Comment by dfinke** [comment]
Yes, that is has been reported https://github.com/dfinke/ImportExcel/discussions/1279
Try v7.4.2. for now
